### ISSUE TITLE
Bundletester doesn't seem to honor controllers

### ISSUE THREAD

**Issue by seman** [juju-solutions/bundletester#93]
This bug was original created under CWR juju-solutions/cloud-weather-report#100 but the actual issue is in BT. `cwr` is passing the correct model to `BT`. This issue is also reproducible using just `BT` instead of `cwr`.

I have 3 controllers/models setup:  gce, azure, and lxd.  My azure controller is currently active:

```
$ juju switch
azure-w:ztest
```

I try to run cwr against gce and lxd (**not azure**), but it starts deploying to azure anyway:

```
$ cwr -F -l DEBUG -v gce-c:ztest lxd:ztest zeppelin.yaml --results-dir /srv/artifacts --test-id 1
...
2017-03-24 17:17:15 DEBUG Bootstrap environment: gce-c:ztest
2017-03-24 17:17:15 DEBUG Connecting to gce-c:ztest...
2017-03-24 17:17:17 DEBUG Connected.
2017-03-24 17:17:17 DEBUG call ['/usr/bin/charm-proof'] (cwd: /tmp/cwr-tmp-9TsCvP/bundletester-3Kf0FH/zeppelin)
2017-03-24 17:17:17 DEBUG Bootstrap environment: lxd:ztest
2017-03-24 17:17:17 DEBUG I: relation mahout has no hooks
2017-03-24 17:17:17 DEBUG I: relation giraph has no hooks
2017-03-24 17:17:17 DEBUG I: File config.yaml not found.
2017-03-24 17:17:18 DEBUG Exit Code: 0
2017-03-24 17:17:18 DEBUG call ['/usr/bin/make', '-s', 'lint'] (cwd: /tmp/cwr-tmp-9TsCvP/bundletester-3Kf0FH/zeppelin)
2017-03-24 17:17:18 DEBUG Reading package lists...
2017-03-24 17:17:18 DEBUG Building dependency tree...
2017-03-24 17:17:18 DEBUG Reading state information...
2017-03-24 17:17:18 DEBUG Connecting to lxd:ztest...
2017-03-24 17:17:18 DEBUG python-pip is already the newest version (8.1.1-2ubuntu0.4).
2017-03-24 17:17:18 DEBUG python-pip set to manually installed.
2017-03-24 17:17:18 DEBUG 0 upgraded, 0 newly installed, 0 to remove and 11 not upgraded.
2017-03-24 17:17:19 DEBUG Connected.
2017-03-24 17:17:19 DEBUG call ['/usr/bin/charm-proof'] (cwd: /tmp/cwr-tmp-9TsCvP/bundletester-mIdA35/zeppelin)
2017-03-24 17:17:19 DEBUG Collecting tox
2017-03-24 17:17:19 DEBUG   Downloading tox-2.6.0-py2.py3-none-any.whl (42kB)
2017-03-24 17:17:20 DEBUG Collecting pluggy<1.0,>=0.3.0 (from tox)
2017-03-24 17:17:20 DEBUG   Downloading pluggy-0.4.0-py2.py3-none-any.whl
2017-03-24 17:17:20 DEBUG I: relation mahout has no hooks
2017-03-24 17:17:20 DEBUG I: relation giraph has no hooks
2017-03-24 17:17:20 DEBUG I: File config.yaml not found.
2017-03-24 17:17:20 DEBUG Collecting py>=1.4.17 (from tox)
2017-03-24 17:17:20 DEBUG Exit Code: 0
2017-03-24 17:17:20 DEBUG call ['/usr/bin/make', '-s', 'lint'] (cwd: /tmp/cwr-tmp-9TsCvP/bundletester-mIdA35/zeppelin)
2017-03-24 17:17:20 DEBUG   Downloading py-1.4.33-py2.py3-none-any.whl (83kB)
2017-03-24 17:17:20 DEBUG Reading package lists...
2017-03-24 17:17:20 DEBUG Requirement already satisfied (use --upgrade to upgrade): virtualenv>=1.11.2; python_version != "3.2" in /usr/lib/python2.7/dist-packages (from tox)
2017-03-24 17:17:20 DEBUG Installing collected packages: pluggy, py, tox
2017-03-24 17:17:20 DEBUG Building dependency tree...
2017-03-24 17:17:20 DEBUG Reading state information...
2017-03-24 17:17:20 DEBUG Successfully installed pluggy-0.4.0 py-1.4.33 tox-2.6.0
2017-03-24 17:17:20 DEBUG python-pip is already the newest version (8.1.1-2ubuntu0.4).
2017-03-24 17:17:20 DEBUG 0 upgraded, 0 newly installed, 0 to remove and 11 not upgraded.
2017-03-24 17:17:20 DEBUG You are using pip version 8.1.1, however version 9.0.1 is available.
2017-03-24 17:17:20 DEBUG You should consider upgrading via the 'pip install --upgrade pip' command.
2017-03-24 17:17:21 DEBUG py34 create: /tmp/cwr-tmp-9TsCvP/bundletester-3Kf0FH/zeppelin/.tox/py34
2017-03-24 17:17:21 DEBUG ERROR: InterpreterNotFound: python3.4
2017-03-24 17:17:21 DEBUG py35 create: /tmp/cwr-tmp-9TsCvP/bundletester-3Kf0FH/zeppelin/.tox/py35
2017-03-24 17:17:21 DEBUG Requirement already satisfied (use --upgrade to upgrade): tox in /usr/local/lib/python2.7/dist-packages
2017-03-24 17:17:21 DEBUG Requirement already satisfied (use --upgrade to upgrade): pluggy<1.0,>=0.3.0 in /usr/local/lib/python2.7/dist-packages (from tox)
2017-03-24 17:17:21 DEBUG Requirement already satisfied (use --upgrade to upgrade): py>=1.4.17 in /usr/local/lib/python2.7/dist-packages (from tox)
2017-03-24 17:17:21 DEBUG Requirement already satisfied (use --upgrade to upgrade): virtualenv>=1.11.2; python_version != "3.2" in /usr/lib/python2.7/dist-packages (from tox)
2017-03-24 17:17:22 DEBUG You are using pip version 8.1.1, however version 9.0.1 is available.
2017-03-24 17:17:22 DEBUG You should consider upgrading via the 'pip install --upgrade pip' command.
2017-03-24 17:17:22 DEBUG py34 create: /tmp/cwr-tmp-9TsCvP/bundletester-mIdA35/zeppelin/.tox/py34
2017-03-24 17:17:22 DEBUG ERROR: InterpreterNotFound: python3.4
2017-03-24 17:17:22 DEBUG py35 create: /tmp/cwr-tmp-9TsCvP/bundletester-mIdA35/zeppelin/.tox/py35
2017-03-24 17:17:26 DEBUG py35 installdeps: -r/tmp/cwr-tmp-9TsCvP/bundletester-3Kf0FH/zeppelin/requirements.txt
2017-03-24 17:17:27 DEBUG py35 installdeps: -r/tmp/cwr-tmp-9TsCvP/bundletester-mIdA35/zeppelin/requirements.txt
2017-03-24 17:17:29 DEBUG py35 installed: appdirs==1.4.3,flake8==3.3.0,mccabe==0.6.1,packaging==16.8,pkg-resources==0.0.0,py==1.4.33,pycodestyle==2.3.1,pyflakes==1.5.0,pyparsing==2.2.0,pytest==3.0.7,six==1.10.0
2017-03-24 17:17:29 DEBUG ___________________________________ summary ____________________________________
2017-03-24 17:17:29 DEBUG SKIPPED:  py34: InterpreterNotFound: python3.4
2017-03-24 17:17:29 DEBUG   py35: skipped tests
2017-03-24 17:17:29 DEBUG   congratulations :)
2017-03-24 17:17:30 DEBUG py35 installed: appdirs==1.4.3,flake8==3.3.0,mccabe==0.6.1,packaging==16.8,pkg-resources==0.0.0,py==1.4.33,pycodestyle==2.3.1,pyflakes==1.5.0,pyparsing==2.2.0,pytest==3.0.7,six==1.10.0
2017-03-24 17:17:30 DEBUG ___________________________________ summary ____________________________________
2017-03-24 17:17:30 DEBUG SKIPPED:  py34: InterpreterNotFound: python3.4
2017-03-24 17:17:30 DEBUG   py35: skipped tests
2017-03-24 17:17:30 DEBUG   congratulations :)
2017-03-24 17:17:31 DEBUG I: relation mahout has no hooks
2017-03-24 17:17:31 DEBUG I: relation giraph has no hooks
2017-03-24 17:17:31 DEBUG I: File config.yaml not found.
2017-03-24 17:17:31 DEBUG Exit Code: 0
2017-03-24 17:17:31 DEBUG call ['/tmp/cwr-tmp-9TsCvP/bundletester-3Kf0FH/zeppelin/tests/01-basic-deployment.py'] (cwd: /tmp/cwr-tmp-9TsCvP/bundletester-3Kf0FH/zeppelin)
2017-03-24 17:17:31 DEBUG I: relation mahout has no hooks
2017-03-24 17:17:31 DEBUG I: relation giraph has no hooks
2017-03-24 17:17:31 DEBUG I: File config.yaml not found.
2017-03-24 17:17:31 DEBUG Exit Code: 0
2017-03-24 17:17:31 DEBUG call ['/tmp/cwr-tmp-9TsCvP/bundletester-mIdA35/zeppelin/tests/01-basic-deployment.py'] (cwd: /tmp/cwr-tmp-9TsCvP/bundletester-mIdA35/zeppelin)
2017-03-24 17:17:32 DEBUG 2017-03-24 17:17:32 Starting deployment of azure-w:ci-70/ztest
2017-03-24 17:17:33 DEBUG 2017-03-24 17:17:33 Starting deployment of azure-w:ci-70/ztest
2017-03-24 17:17:34 DEBUG 2017-03-24 17:17:34 Deploying applications...
2017-03-24 17:17:35 DEBUG 2017-03-24 17:17:35 Deploying applications...
```

I think this is the root cause for https://github.com/juju-solutions/layer-cwr/issues/114.

Edit to include pip versions:

```bash
root@ham:~# pip list | grep cloud-weather
cloud-weather-report (0.4.2)

root@ham:~# pip list | grep bundle
bundletester (0.11.0)
jujubundlelib (0.5.2)

root@ham:~# pip3 list | grep bundle
jujubundlelib (0.5.3)
```

Full log of the test (which is still running):

http://paste.ubuntu.com/24242001/

The top shows it's thinking about running on gce and lxd, but it's deploying to azure.

**Comment by seman** [comment]
This is Bundletester issue. Moving it to juju-solutions/bundletester#93

**Comment by seman** [comment]
@kwmonroe can you try this with this version of amulet https://github.com/seman/amulet/tree/juju_model ?

**Comment by seman** [comment]
Closed via #94
